src/plugins/intel_npu/src/common/include/intel_npu/common/igraph.hpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include "intel_npu/icompiler.hpp"
```

Comment:
icompiler.hpp should only be used by the PluginCompilerAdapter now. I think we need it here only for NetworkMetadata now, but igraph.hpp should define its own Metadata struct and PluginCompilerAdapter should be the only one making the translation from icompiler.hpp to igraph.hpp format.

Reply:
Correction: it is not that easy to remove the usage of NetworkMetadata from all layers except the PluginCompilerAdapter. We might need to do it in a separate PR ...

Reply:
we created another header for metadata and that is used instead of icompiler.